libmachine/host.go

```diff
@@ -256,6 +256,15 @@ func (h *Host) Upgrade() error {
 }
 
 func (h *Host) Remove(force bool) error {
+	machineState, err := h.Driver.GetState()
+	if err != nil {
+		return err
+	}
+
+	if machineState != state.Stopped {
```

Comment:
Well, I'm not sure what you mean by "automatically", but if a machine is in a permanent `Error`-state (maybe because it was removed from the provider, or if the account on the provider is gone, or whatnot), there needs to be a way to remove the host from docker-machine... Well, other than deleting it from `~/.docker/machine/machines/`, that is :P